cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -396,21 +411,33 @@ def _update_list_response(self):
         """
         Sync the local cache of all firewalls with the server.
         """
-        self._list_response = (
-                self._provider.gce_compute
-                              .firewalls()
-                              .list(project=self._provider.project_name)
-                              .execute())
+        self._list_response = None
+        token = None
+        while True:
+            list_response = (self._provider
+                                 .gce_compute
+                                 .firewalls()
+                                 .list(project=self._provider.project_name,
+                                       pageToken=token)
+                                 .execute())
+            if self._list_response is None:
+                self._list_response = list_response
+                if 'items' not in self._list_response:
+                    self._list_response['items'] = []
+            else:
+                self._list_response['items'] += list_response.get('items', [])
+            if 'nextPageToken' not in list_response:
+                return
+            token = list_response['nextPageToken']
```

Comment:
@chiniforooshan I feel that there are several places where this overall pattern of logic is being repeated. I wonder whether it can be factored out into a helper like with the PageableObjectMixin: https://github.com/gvlproject/cloudbridge/blob/master/cloudbridge/cloud/base/resources.py#L200

So that teration can simply be something like:
```
for item in gce_helpers.iter_all(self._provider.gce_compute.firewalls(), project=blah, pageToken=token):
   do something
```

and iter_all could be something like:
```
def iter_all(gce_resource, **kwargs):
    do:
       response =   gce_resource.list(kwargs)
       if 'items' not in response:
             return
       for item in response['items']:
              yield item
        token = response.get('nextPageToken')
    while token
```